pyccel/codegen/printing/ccode.py

```diff
@@ -282,7 +284,9 @@ def copy_NumpyArray_Data(self, expr):
         arg = rhs.arg
         if rhs.rank > 1:
             # flattening the args to use them in C initialization.
-            arg = functools.reduce(operator.concat, arg)
+            flatten_list = lambda irregular_list:[element for item in irregular_list for element in flatten_list(item)] if isinstance(irregular_list, (PythonList, PythonTuple)) else [irregular_list]
```

Comment:
Is there a reason for this to be a lambda rather than a local function? In any case it needs breaking up into several lines as it does not fit on my screen